lib/chef/resource/homebrew_install.rb

```diff
@@ -0,0 +1,147 @@
+      ```ruby
+      homebrew_install 'Install Homebrew and xcode command line tools if necessary' do
+        user 'someuser'
+        action :install
+      end
+      ```
+      **Install Homebrew using a local source to download Command Line Tools for Xcode from**:
+      ```ruby
+      homebrew_install 'Install Homebrew and xcode command line tools if necessary' do
+        tools_url 'https://somewhere.something.com/downloads/command_line_tools.dmg'
+        tools_pkg_name 'Command Line Tools.pkg'
+        user 'someuser'
+        action :install
+      end
+      ```
+      DOC
+
+      property :tools_url, String,
+        description: "A url pointing to a local source for the Command Line Tools for Xcode dmg"
+
+      property :tools_pkg_name, String,
+        description: "The name of the pkg inside the dmg located at the tools url"
+
+      property :brew_source, String,
+        description: "A url pointing to a Homebrew installer",
+        default: "https://github.com/Homebrew/brew/tarball/master"
+
+      property :user, String,
+        description: "The user to install Homebrew as. Note: Homebrew cannot be installed as root.",
+        required: true
+
+      action :install do
+        # Avoid all the work in the below resources if homebrew is already installed
+        return if ::File.exist?("/usr/local/bin/brew")
+
+        BREW_REPO = "https://codeload.github.com/Homebrew/brew/zip/master".freeze
+        USER_HOME = Dir.home(new_resource.user).freeze
+        HOMEBREW_CACHE = "#{USER_HOME}/Library/Caches/Homebrew".freeze
+
+        # Creating the basic directory structure needed for Homebrew
+        directories = ["bin", "etc", "include", "lib", "sbin", "share", "var", "opt",
+                        "share/zsh", "share/zsh/site-functions",
+                        "var/homebrew", "var/homebrew/linked",
+                        "Cellar", "Caskroom", "Homebrew", "Frameworks"
+                      ].freeze
+        directories.each do |dir|
+          directory "/usr/local/#{dir}" do
+            mode "0755"
+            owner new_resource.user
+            group "admin"
+            action :create
+          end
+        end
+
+        user_directories = ["#{USER_HOME}", "#{USER_HOME}/Library",
+                            "#{USER_HOME}/Library/Caches", "#{USER_HOME}/Library/Caches/Homebrew"
+        ]
+        user_directories.each do |dir|
+          directory "#{dir}" do
+            mode "0755"
+            owner new_resource.user
+            group "admin"
+            action :create
+          end
+        end
+
+        if new_resource.tools_url
+          dmg_package new_resource.tools_pkg_name do
+            source new_resource.tools_url
+            type "pkg"
+          end
+        else
+          build_essential "install Command Line Tools for Xcode" do
+            action :upgrade
+          end
+        end
+
+        script "Download and unpack Homebrew" do
+          interpreter "bash"
```

Comment:
Any reason not to make this a `bash` resource?

Reply:
as discussed, will file a bug